homeassistant/components/lamarzocco/water_heater.py

```diff
@@ -0,0 +1,175 @@
+
+
+@dataclass
+class LaMarzoccoWaterHeaterEntityDescription(
+    WaterHeaterEntityEntityDescription,
+    LaMarzoccoEntityDescription,
+    LaMarzoccoWaterHeaterEntityDescriptionMixin,
+):
+    """Description of an La Marzocco Water Heater."""
+
+
+ENTITIES: tuple[LaMarzoccoWaterHeaterEntityDescription, ...] = (
+    LaMarzoccoWaterHeaterEntityDescription(
+        key="coffee_boiler",
+        translation_key="coffee_boiler",
+        icon="mdi:coffee-maker",
+        min_temp=85,
+        max_temp=104,
+        set_temp_fn=lambda client, temp: client.set_coffee_temp(temp),
+        current_op_fn=lambda client: client.current_status.get("power", False),
+        control_fn=lambda client, state: client.set_power(state),
+        current_temp_fn=lambda client: client.current_status.get("coffee_temp", 0),
+        target_temp_fn=lambda client: client.current_status.get("coffee_temp_set", 0),
+        extra_attributes={},
+    ),
+    LaMarzoccoWaterHeaterEntityDescription(
+        key="steam_boiler",
+        translation_key="steam_boiler",
+        icon="mdi:kettle-steam",
+        min_temp=126,
+        max_temp=131,
+        set_temp_fn=lambda client, temp: client.set_steam_temp(round(temp)),
+        current_op_fn=lambda client: client.current_status.get(
+            "steam_boiler_enable", False
+        ),
+        control_fn=lambda client, state: client.set_steam_boiler_enable(state),
+        current_temp_fn=lambda client: client.current_status.get("steam_temp", 0),
+        target_temp_fn=lambda client: client.current_status.get("steam_temp_set", 0),
+        extra_attributes={},
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up water heater type entities."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        LaMarzoccoWaterHeater(coordinator, hass, description)
+        for description in ENTITIES
+        if not description.extra_attributes
+        or coordinator.lm.model_name in description.extra_attributes
+    )
+
+
+class LaMarzoccoWaterHeater(LaMarzoccoEntity, WaterHeaterEntity):
+    """Water heater representing espresso machine temperature data."""
+
+    entity_description: LaMarzoccoWaterHeaterEntityDescription
+
+    @property
+    def supported_features(self) -> WaterHeaterEntityFeature:
+        """Return the list of supported features."""
+        return (
+            WaterHeaterEntityFeature.TARGET_TEMPERATURE
+            | WaterHeaterEntityFeature.ON_OFF
+            | WaterHeaterEntityFeature.OPERATION_MODE
+        )
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        return UnitOfTemperature.CELSIUS
+
+    @property
+    def precision(self) -> float:
+        """Return the precision of the platform."""
+        return PRECISION_TENTHS
```

Comment:
Please use short hand notation as class variable similar to `_attr_has_entity_name = True`